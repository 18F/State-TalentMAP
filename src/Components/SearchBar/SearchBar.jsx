import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchText: { value: '' },
    };
  }
  changeText(e) {
    const { searchText } = this.state;
    searchText.value = e.target.value;
    this.setState({ searchText }, this.props.onChangeText(e));
  }
  render() {
    const hidden = {
      display: 'none',
    };
    const { id, type, submitDisabled, submitText, alertText, onSubmitSearch, label, labelSrOnly }
      = this.props;
    const { searchText } = this.state;
    let showSubmitText = true; // do not hide submit text initially
    if (type === 'small') { showSubmitText = false; } // small search class should not have text
    return (
      <div className={`usa-search usa-search-${type}`}>
        <div role="search">
          <form onSubmit={e => onSubmitSearch(e)}>
            <label className={labelSrOnly ? 'usa-sr-only' : null} htmlFor={id}>
              {label}
            </label>
            <input
              id={id}
              value={searchText.value}
              onChange={e => this.changeText(e)}
              type="search"
              name="search"
            />
            <div id="enabled-search">
              <button
                id="enabled-search-button"
                className={submitDisabled ? 'usa-button-disabled' : null}
                disabled={submitDisabled}
                type="submit"
              >
                <span className="usa-search-submit-text">{showSubmitText ? submitText : null}</span>
              </button>
            </div>
            <div id="disabled-search" style={hidden}>
              <button
                className="usa-button-disabled"
                disabled="true"
                type="submit"
                id="disabled-search-button"
              >
                <span className="usa-search-submit-text usa-button-disabled">
                  {showSubmitText ? submitText : null}
                </span>
              </button>
              <span className="alert-text">{alertText}</span>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

SearchBar.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  type: PropTypes.oneOf(['small', 'medium', 'big']),
  submitDisabled: PropTypes.bool,
  submitText: PropTypes.string.isRequired,
  alertText: PropTypes.string,
  onChangeText: PropTypes.func.isRequired,
  onSubmitSearch: PropTypes.func.isRequired,
  labelSrOnly: PropTypes.bool,
};

SearchBar.defaultProps = {
  type: 'big', // should be one of the USWDS search types - https://standards.usa.gov/components/search-bar/
  submitDisabled: false,
  alertText: 'Disabled',
  label: 'Search', // sr only if flagged
  labelSrOnly: true,
};

export default SearchBar;
