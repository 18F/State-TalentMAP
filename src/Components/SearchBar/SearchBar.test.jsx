import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import SearchBar from './SearchBar';

['small', 'medium', 'big'].forEach((size) => {
  describe(`SearchBarComponent ${size}`, () => {
    let wrapper = null;

    it('can receive props', () => {
      wrapper = shallow(
        <SearchBar
          id="search"
          type={size}
          submitDisabled
          submitText="Submit"
          alertText="Search disabled"
          onChangeText={() => {}}
          onSubmitSearch={() => {}}
        />,
      );
      expect(wrapper.instance().props.id).toBe('search');
    });

    it('handles different props', () => {
      wrapper = shallow(
        <SearchBar
          id="search-2"
          label="Label"
          type={size}
          submitDisabled={false}
          submitText="Submit 2"
          alertText="Search is disabled"
          onChangeText={() => {}}
          onSubmitSearch={() => {}}
        />,
      );
      expect(wrapper.instance().props.id).toBe('search-2');
    });

    it('can change text', () => {
      wrapper = shallow(
        <SearchBar
          id="search-2"
          type={size}
          submitDisabled={false}
          submitText="Submit 2"
          alertText="Search is disabled"
          onChangeText={() => {}}
          onSubmitSearch={() => {}}
        />,
      );
      wrapper.instance().changeText({ target: { value: 'test' } });
      expect(wrapper.instance().state.searchText.value).toBe('test');
    });

    it('matches snapshot', () => {
      wrapper = shallow(
        <SearchBar
          id="search-2"
          label="Label"
          type={size}
          submitDisabled={false}
          submitText="Submit 2"
          alertText="Search is disabled"
          onChangeText={() => {}}
          onSubmitSearch={() => {}}
        />,
      );
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
