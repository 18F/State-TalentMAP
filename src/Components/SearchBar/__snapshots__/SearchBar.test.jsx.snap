// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarComponent big matches snapshot 1`] = `
<div
  className="usa-search usa-search-big"
>
  <div
    role="search"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="label-input-wrapper"
      >
        <label
          className="usa-sr-only"
          htmlFor="search-2"
        >
          Label
        </label>
        <input
          id="search-2"
          name="search"
          onChange={[Function]}
          placeholder={null}
          type="search"
          value=""
        />
        <div
          id="enabled-search"
        >
          <button
            className={null}
            disabled={false}
            id="enabled-search-button"
            type="submit"
          >
            <span
              className="usa-search-submit-text"
            >
              Submit 2
            </span>
            <span
              className="usa-sr-only"
            >
              Submit 2
            </span>
          </button>
        </div>
        <div
          id="disabled-search"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <button
            className="usa-button-disabled"
            disabled="true"
            id="disabled-search-button"
            type="submit"
          >
            <span
              className="usa-search-submit-text usa-button-disabled"
            >
              Submit 2
            </span>
          </button>
          <span
            className="alert-text"
          >
            Search is disabled
          </span>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`SearchBarComponent medium matches snapshot 1`] = `
<div
  className="usa-search usa-search-medium"
>
  <div
    role="search"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="label-input-wrapper"
      >
        <label
          className="usa-sr-only"
          htmlFor="search-2"
        >
          Label
        </label>
        <input
          id="search-2"
          name="search"
          onChange={[Function]}
          placeholder={null}
          type="search"
          value=""
        />
        <div
          id="enabled-search"
        >
          <button
            className={null}
            disabled={false}
            id="enabled-search-button"
            type="submit"
          >
            <span
              className="usa-search-submit-text"
            >
              Submit 2
            </span>
            <span
              className="usa-sr-only"
            >
              Submit 2
            </span>
          </button>
        </div>
        <div
          id="disabled-search"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <button
            className="usa-button-disabled"
            disabled="true"
            id="disabled-search-button"
            type="submit"
          >
            <span
              className="usa-search-submit-text usa-button-disabled"
            >
              Submit 2
            </span>
          </button>
          <span
            className="alert-text"
          >
            Search is disabled
          </span>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`SearchBarComponent small matches snapshot 1`] = `
<div
  className="usa-search usa-search-small"
>
  <div
    role="search"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="label-input-wrapper"
      >
        <label
          className="usa-sr-only"
          htmlFor="search-2"
        >
          Label
        </label>
        <input
          id="search-2"
          name="search"
          onChange={[Function]}
          placeholder={null}
          type="search"
          value=""
        />
        <div
          id="enabled-search"
        >
          <button
            className={null}
            disabled={false}
            id="enabled-search-button"
            type="submit"
          >
            <span
              className="usa-search-submit-text"
            />
            <span
              className="usa-sr-only"
            >
              Search
            </span>
          </button>
        </div>
        <div
          id="disabled-search"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <button
            className="usa-button-disabled"
            disabled="true"
            id="disabled-search-button"
            type="submit"
          >
            <span
              className="usa-search-submit-text usa-button-disabled"
            />
          </button>
          <span
            className="alert-text"
          >
            Search is disabled
          </span>
        </div>
      </div>
    </form>
  </div>
</div>
`;
